azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -781,3 +790,42 @@ def _remove_secrets_from_secret_sync(
                     resource=secret_sync,
                 )
                 return wait_for_terminal_state(poller)
+
+    def _get_secret_names(self, keyvault_name: str) -> List[str]:
+        secret_iteratable = self.keyvault_client.get_secrets(
+            vault_base_url=KEYVAULT_URL.format(keyvaultName=keyvault_name)
+        )
+        return [secret.get("id") for secret in secret_iteratable]
```

Comment:
When does a secret not have an id ?

Reply:
If id is not guaranteed this syntax could filter out `None`'s 

```
[secret["id"] for secret in secret_iteratable if "id" in secret]
```

Reply:
theoretically the secret should all have id, but I did not found documentation stating this property as required or optional, therefore just to be safe from unnecessary error, tho I will add filter that you suggested😁

Reply:
Did you push your change? looks the same as before

Reply:
I do see the difference😋should be updated
<img width="327" alt="image" src="https://github.com/user-attachments/assets/8a535652-cd00-4677-a0ce-fcb31112243d">